docs/guide/async.asciidoc

```diff
@@ -0,0 +1,127 @@
+[[async]]
+== Using with asyncio
+
+The `elasticsearch` package supports async/await with
+https://docs.python.org/3/library/asyncio.html[asyncio] and
+https://docs.aiohttp.org[aiohttp]. You can either install `aiohttp`
+directly or use the `[async]` extra:
+
+[source,bash]
+----
+$ python -m pip install elasticsearch aiohttp
+
+# - OR -
+
+$ python -m pip install elasticsearch[async]
+----
+
+=== Getting Started with Async
+
+After installation all async API endpoints are available via
+`~elasticsearch.AsyncElasticsearch` and are used in the same way as
+other APIs, just with an extra `await`:
```

Comment:
I will try not to nitpick on the existing documentation, but as a follow-up, we should remove "just" here.